extras/rya.streams/api/src/test/java/org/apache/rya/streams/api/queries/InMemoryQueryRepositoryTest.java

```diff
@@ -144,6 +171,33 @@ public void update() throws Exception {
             final Optional<StreamsQuery> fetched = queries.get(query.getQueryId());
             final StreamsQuery expected = new StreamsQuery(query.getQueryId(), query.getSparql(), false);
             assertEquals(expected, fetched.get());
+        } finally {
+            queries.stop();
+        }
+    }
+
+    @Test
+    public void updateListenerNotify() throws Exception {
```

Comment:
Could you make a multi-client test that ensures Client A changes will be visible to a listener that is subscribed to Client B, where both are backed by the same change log? This test alone does not verify that works since it's not even starting the repository. The add(query) call is what's triggering the listener update.